packages/block-editor/src/hooks/spacing.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * WordPress dependencies
+ */
+import { getBlockSupport } from '@wordpress/blocks';
+import { Platform } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { PaddingEdit, useIsPaddingDisabled } from './padding';
+import SpacingPanelControl from '../components/spacing-panel-control';
+
+export const SPACING_SUPPORT_KEY = 'spacing';
+
+/**
+ * Inspector controls for spacing support.
+ *
+ * @param  {Object} props Block props.
+ * @return {WPElement}    Inspector controls for spacing support features.
+ */
+export function SpacingPanel( props ) {
+	const isDisabled = useIsSpacingDisabled( props );
+	const isSupported = hasSpacingSupport( props.name );
+
+	if ( isDisabled || ! isSupported ) {
+		return null;
+	}
+
+	return (
+		<SpacingPanelControl key="spacing">
+			<PaddingEdit { ...props } />
+		</SpacingPanelControl>
+	);
+}
+
+/**
+ * Determine whether there is block support for padding.
+ *
+ * @param {string} blockName Block name.
+ * @return {boolean}         Whether there is support.
+ */
+export function hasSpacingSupport( blockName ) {
```

Comment:
Looks like this could use the `hasPaddingSupport` from the padding.js file? like done with `useIsPaddingDisabled`

Reply:
I had this separate so in the next PR all that was needed was the [addition of `|| support?.margin`](https://github.com/WordPress/gutenberg/pull/30608/files#diff-e819cbce5f99d1503f31d9cdae0f87b00c8e76315f683e98b2cde792f92379d8R51)

I could replace this approach with calls to `hasPaddingSupport` and `hasMarginSupport` if that is better. Whatever you'd prefer is fine by me.

Reply:
I don't mind either, I just thought since you shared the hook maybe it makes sense to do the same here; But I don't feel strongly.

Reply:
I've updated this to use `hasPaddingSupport`, it will be quicker to grok now.